### PR TITLE
Akka 2.5.0 news

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: news/_posts/2017-04-13-akka-2.5.0-released.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    * Generic type for the key in Distributed Data ORMap, thanks to [jgordijn](https://github.com/jgordijn)
+* [Coordinated shutdown](http://doc.akka.io/docs/akka/2.5/scala/actors.html#Coordinated_Shutdown) to facilitate ordered stopping of actors and services and perform registered tasks during the shutdown process. This greatly simplifies tasks involved in graceful leaving of Akka Cluster.
+* Support [remembering entities](http://doc.akka.io/docs/akka/2.5/scala/cluster-sharding.html#Remembering_Entities) in Cluster Sharding with the Distributed Data (ddata) mode. ddata is now also the [default mode](http://doc.akka.io/docs/akka/2.5/scala/cluster-sharding.html#Distributed_Data_vs__Persistence_Mode) for Cluster Sharding.
+* New Java API for Testkit taking advantage of lambas, thanks to [@Hawstein](https://github.com/Hawstein)
+* Fix performance regression for Scala 2.12, due to change in scheduling fairness of the JDK8 ForkJoinPool implementation
+* We added @ApiMayChange and [other annotations](https://github.com/akka/akka/pull/22110/files) which can be used to mark experimental API inside stable modules, primarily for use in Akka HTTP.
```
Comment (ktoso) [inline news/_posts/2017-04-13-akka-2.5.0-released.md]:
```
`@ApiMayChange`
```

Comment (patriknw) [inline news/_posts/2017-04-13-akka-2.5.0-released.md]: fixed

File: news/_posts/2017-04-13-akka-2.5.0-released.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+### Binary Compatibility
+
+Akka 2.5.x is backwards binary [compatible](http://doc.akka.io/docs/akka/2.5/common/binary-compatibility-rules.html) with previous 2.4.x versions (exceptions listed below). This means that the new JARs are a drop-in replacement for the old one (but not the other way around) as long as your build does not enable the inliner (Scala-only restriction). It should be noted that Scala 2.12.x is is not binary compatible with Scala 2.11.x, which means that Akka’s binary compatibility property only holds between versions that were built for a given Scala version—akka-actor_2.12-2.5.0.jar is compatible with akka-actor_2.12-2.4.17.jar but not with akka-actor_2.11-2.4.17.jar.
+
+Binary compatibility is *not* maintained for the following:
+
+* *akka-testkit* and *akka-multi-node-testkit*
+* modules and API marked as ["may change](https://github.com/akka/akka/blob/master/akka-actor/src/main/java/akka/annotation/ApiMayChange.java)["](https://github.com/akka/akka/blob/master/akka-actor/src/main/java/akka/annotation/ApiMayChange.java), such as *akka-typed* and *akka-contrib*
```
Comment (raboof) [inline news/_posts/2017-04-13-akka-2.5.0-released.md]: some duplication here?

File: news/_posts/2017-04-13-akka-2.5.0-released.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+* features, classes, methods that were deprecated in 2.4.0 or earlier and removed in 2.5.x
+
+We had to make two additional exceptions:
+
+* Custom Mailbox implementations. Scala 2.12 is using the standard JDK8 ForkJoinPool, which may cause performance regression for some Actor messaging scenarios. Therefore we have embedded the ForkJoinPool from Scala 2.11 in Akka. This breaks binary backwards compatibility for custom Mailbox implementations that were compiled with Akka 2.4.
+* To make the Java APIs AbstractPersistentActor and AbstractPersistentActorWithAtLeastOnceDelivery to work with Scala 2.12 we could find no other solution but to break the binary compatibility of the Scala versions (which the Java ones were based on).
```
Comment (raboof) [inline news/_posts/2017-04-13-akka-2.5.0-released.md]: Perhaps add some backticks here for classes?

**Pull request merged**